src/python/WMCore/MicroService/MSPileup/MSPileupTasks.py

```diff
@@ -262,7 +263,8 @@ def partialPileupTask(self):
                     self.logger.info("Finally, updating the pileup document for pileup name: %s", doc['pileupName'])
 
                     # update MSPileup document in MongoDB
-                    self.mgr.updatePileup(doc, rseList=doc['currentRSEs'])
+                    spec = {'pileupName': doc['pileupName']}
+                    self.mgr.updatePileupDocumentInDatabase(spec, doc, rseList=doc['currentRSEs'])
```

Comment:
we must validate here since we update document with new RSEs, creates custom name, set new fraction, etc. Bottom line too many parameters are updated which may be (potentially) set up wrong way. It is better to be safe through the validation.

Reply:
Understood! Maybe add a comment line above it(?)

Reply:
I added comments to the code.